packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -163,12 +206,51 @@ export class Table extends Construct {
     }
 
     public addPartitionKey(attribute: Attribute): this {
-        this.addKey(attribute.name, attribute.type, HASH_KEY_TYPE);
+        this.addKey(attribute, HASH_KEY_TYPE);
         return this;
     }
 
     public addSortKey(attribute: Attribute): this {
-        this.addKey(attribute.name, attribute.type, RANGE_KEY_TYPE);
+        this.addKey(attribute, RANGE_KEY_TYPE);
+        return this;
+    }
+
+    public addGlobalSecondaryIndex(props: SecondaryIndexProps): this {
+        if (this.globalSecondaryIndexes.length === 5) {
+            // https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html#limits-secondary-indexes
+            throw new RangeError('a maximum number of global secondary index per table is 5');
+        }
+
+        if (props.projectionType === ProjectionType.Include && !props.nonKeyAttributes) {
+            // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-projectionobject.html
+            throw new Error(`nonKeyAttributes must be specified when using ${ProjectionType.Include} projection type`);
+        }
```

Comment:
Is also true the other way around? Should we add a validation that if nonKeyAttributes is set the projectType is not KEYS-ONLY?

Reply:
I suppose that you meant `Include`, not `KEYS-ONLY`.

I think adding validation makes sense from use cases perspective. For example, `ALL` with non-key attributes does not give any value. I will check behavior and update this snippet accordingly.